augur/io/sqlite3.py

```diff
@@ -0,0 +1,137 @@
+import os
+import pandas as pd
+import sqlite3
+from typing import List
+
+
+class Sqlite3Database:
+    def __init__(self, file:str=''):
+
+        self.file = file
+        """Database file. Empty string if using an in-memory database."""
+
+        self.use_in_memory_db = not self.file
+        """Whether to use an in-memory database."""
+
+        if self.use_in_memory_db:
+            # More info: https://www.sqlite.org/inmemorydb.html
+            self.in_memory_db_connection = sqlite3.connect(":memory:")
+            """Persistent connection to an in-memory database."""
+
+    def connect(self, **connect_kwargs) -> sqlite3.Connection:
+        """Return a connection to the SQLite database.
+        If using an in-memory database, the same connection is returned for all calls to this method.
+        Otherwise, a new connection is returned every time."""
+        if self.use_in_memory_db:
+            return self.in_memory_db_connection
+        return sqlite3.connect(self.file, **connect_kwargs)
+
+    def execute(self, *args):
+        """Execute under a connection context."""
+        with self.connect() as con:
+            con.execute(*args)
+
+    def executemany(self, *args):
+        """Executemany under a connection context."""
+        # TODO: benchmark performance with cursor.executemany / con.commit at the end
+        with self.connect() as con:
+            con.executemany(*args)
+
+    def create_unique_index(self, table_name:str, column:str):
+        """Create a unique index on a column in a table."""
+        index_name = f'index_{table_name}_{column}'
+        self.execute(f"""
+            CREATE UNIQUE INDEX {sanitize_identifier(index_name)}
+            ON {sanitize_identifier(table_name)} ({sanitize_identifier(column)})
```

Comment:
Is the need to sanitize identifiers here unique to the creation of an index in sqlite3? Why do we not sanitize ids like table name or column elsewhere?